Add Tailwind to advance options.

PrajvalBadiger opened the pull request:
By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

## Problem/Feature
Resolves #186 

## Description of Changes: 
This PR adds an option to the `--advanced` flag that let's users add Tailwind CSS framework to the project
- This change creates package.json, tailwind.config.js, input.css and output.css files
- pnpm is used to install tailwind via package.json as I couldn't test with npm
- updated Makefile to run tailwind for the build recipe. which is also used in .air.toml

## Checklist

- [x] I have self-reviewed the changes being requested
- [x] I have updated the documentation (if applicable)

briancbarrow commented:
@PrajvalBadiger Thanks for submitting this PR. I wanted to ask what you thought about changing the implementation to use the [Tailwind Standalone CLI](https://tailwindcss.com/blog/standalone-cli) instead of installing the package with Node.js? 

I've used the standalone CLI in my projects and it works well. We'd have to add some language to the output saying to make sure they have it installed, but we are already doing something similar with Templ so I don't think that will be an issue. 

The other thing we probably want to add is some checks to make sure that the tailwind option is not used without the htmx option.

PrajvalBadiger commented:
> @PrajvalBadiger Thanks for submitting this PR. I wanted to ask what you thought about changing the implementation to use the [Tailwind Standalone CLI](https://tailwindcss.com/blog/standalone-cli) instead of installing the package with Node.js?
> 
> I've used the standalone CLI in my projects and it works well. We'd have to add some language to the output saying to make sure they have it installed, but we are already doing something similar with Templ so I don't think that will be an issue.
> 

@briancbarrow, Thanks for the review. I was not aware of the Tailwind Standalone CLI and had used Node.js for Tailwind in my projects. A huge dependency on Node.js would be reduced by using the standalone CLI.

> The other thing we probably want to add is some checks to make sure that the tailwind option is not used without the htmx option.

There a few ways we can implement this:
1. By Default selecting the tailwind option is disabled and user can only select tailwind option if htmx option is selected
2. If tailwind option is selected, htmx is automatically selected
3. If only the tailwind option is used without htmx option, the program fails and print an error. If this is the case the user has to redo all the steps and select htmx option with tailwind option.

briancbarrow commented:
> There a few ways we can implement this:

> By Default selecting the tailwind option is disabled and user can only select tailwind option if htmx option is selected
> If tailwind option is selected, htmx is automatically selected
> If only the tailwind option is used without htmx option, the program fails and print an error. If this is the case the user has to redo all the steps and select htmx option with tailwind option.

I would be good with either the first or second option.

PrajvalBadiger commented:
1. I have implemented the second option - If tailwind option is selected, htmx is automatically selected.
2. Node.js dependency is droped and tailwind standalone cli can be used now.
3. Added note to remind users to install tailwind standalone cli.

briancbarrow commented on cmd/create.go:
```diff
@@ -244,6 +244,12 @@ var createCmd = &cobra.Command{
+		if options.Advanced.Choices["Tailwind"] {
+			options.Advanced.Choices["Htmx"] = true
+			fmt.Println(endingMsgStyle.Render("• Install the tailwind standalone cli if you haven't already, grab the executable for your platform from the latest release on GitHub\n"))
+			fmt.Println(endingMsgStyle.Render("• Tailwind GitHub: https://github.com/tailwindlabs/tailwindcss/releases/\n"))
```
I would change this to point to the page explaining the Standalone CLI. That page will give more info about it and how to install it. It is still a relatively unknown feature so pointing people to an explainer page will be helpful. 
```suggestion
			fmt.Println(endingMsgStyle.Render("• More info about the Tailwind CLI: https://tailwindcss.com/blog/standalone-cli\n"))
```

briancbarrow commented on cmd/steps/steps.go:
```diff
@@ -108,6 +108,11 @@ func InitSteps(projectType flags.Framework, databaseType flags.Database) *Steps
+					{
+						Flag:  "Tailwind",
+						Title: "TailwindCSS",
+						Desc:  "A utility-first CSS framework",
```
```suggestion
						Desc:  "A utility-first CSS framework (selecting this will automatically add HTMX)",
```

briancbarrow reviewed (changes requested):
This is looking great. Some small changes to make things clearer in the output of the cli.

briancbarrow commented:
Thanks @PrajvalBadiger! The only thing left is it looks like there are commits on here that are from previously merged PRs. Can you do a rebase and only keep the commits that are relevant to this PR?

Melkeydev commented on README.md:
```diff
@@ -111,6 +111,9 @@ You can now use the `--advanced` flag when running the `create` command to get a
 - [Websocket](https://pkg.go.dev/nhooyr.io/websocket) sets up a websocket endpoint
+- [Tailwind](https://tailwindcss.com/) Css framework
+
+Note: selecting tailwind option automatically selects htmx.
```
nice

Melkeydev commented on cmd/program/program.go:
```diff
@@ -216,7 +216,7 @@ func (p *Project) CreateMainFile() error {
 	if _, err := os.Stat(p.AbsolutePath); os.IsNotExist(err) {
 		// create directory
-		if err := os.Mkdir(p.AbsolutePath, 0754); err != nil {
+		if err := os.Mkdir(p.AbsolutePath, 0o754); err != nil {
```
What is this doing?

PrajvalBadiger replied:
This is nothing to do with the PR itself, go lsp in neovim auto formats octals 0o.

briancbarrow reviewed (approved):
LGTM
